Tasks/Common/PowershellHelpers/Invoke-ActionWithRetries.ps1

```diff
@@ -0,0 +1,109 @@
+function Invoke-ActionWithRetries {
+    [CmdletBinding()]
+    param(
+        [scriptblock]
+        $Action,
+
+        [scriptblock]
+        $ActionSuccessValidator = { $true },
+
+        [int32]
+        $MaxTries = 10,
+
+        [int32]
+        $RetryIntervalInSeconds = 1,
+
+        [string[]]
+        [ValidateScript({[System.Exception].IsAssignableFrom([type]$_)})]
+        $RetryableExceptions,
+
+        [switch]
+        $ContinueOnError = $false,
+
+        [string]
+        $RetryMessage
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+
+    if(!$RetryMessage)
+    {
+        $RetryMessage = Get-VstsLocString -Key RetryAfterMessage $RetryIntervalInSeconds
+    }
+
+    $retryIteration = 1
+    do
+    {
+        $shouldRetry = $false
+        $result = $false
+        $exception = $null
+
+        try
+        {
+            $result = & $Action
+        }
+        catch
+        {
+            if(($null -eq $RetryableExceptions) -or (Test-RetryableException -Exception $_.Exception -AllowedExceptions $RetryableExceptions))
+            {
+                $exception = $_.Exception
+            }
+            else
+            {
+                throw
+            }
+        }
+
+        if(!$exception -and (!$result -or $ActionSuccessValidator.Invoke($result)))
+        {
+            return $result
+        }
+
+        $shouldRetry = $true
+
+        if($retryIteration -eq $MaxTries)
+        {
+            if(!$ContinueOnError)
```

Comment:
removed it as we are not using it. Although it would have been nice to have, it is equally easy for caller to handle whether or not to continue on error.